### PR TITLE
Added userLabels for usage report

### PR DESCRIPTION
**Pull request by arshan-alam**

### CODE REVIEW

File: java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java [hunk 1]
```diff
@@ -47,15 +50,11 @@ public static void main(String[] args) throws Exception {
...
-    ServiceControl serviceControl =
-        new ServiceControl.Builder(
-                GoogleNetHttpTransport.newTrustedTransport(),
-                new JacksonFactory(),
-                new HttpCredentialsAdapter(
-                    GoogleCredentials.getApplicationDefault().createScoped(
-                        Collections.singletonList(CLOUD_SCOPE))))
-            .setApplicationName(ServiceOptions.getDefaultProjectId())
-            .build();
+    ServiceControl serviceControl = new ServiceControl.Builder(GoogleNetHttpTransport.newTrustedTransport(),
```
Comment (tangbiao) [inline java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java:53]: Don't see any difference. Is it just the format?

File: java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java [hunk 2]
```diff
@@ -65,10 +64,8 @@ public static void main(String[] args) throws Exception {
           continue;
         }
 
-        String startTime =
-            product.lastReportTime != null && !product.lastReportTime.isEmpty()
-                ? product.lastReportTime
-                : product.startTime;
+        String startTime = product.lastReportTime != null && !product.lastReportTime.isEmpty() ? product.lastReportTime
```
Comment (tangbiao) [inline java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java:67]: ditto, and a few places below

File: python3/impl/step_6_usage_reporting/report.py [hunk 3]
```diff
@@ -86,6 +86,13 @@ def main(argv):
                 print(check['checkErrors'])
                 ### TODO: Temporarily turn off service for the user. ###
                 continue
+
+            # userLabels are only allowed in report()
+            operation['userLabels'] = {
```
Comment (tangbiao) [inline python3/impl/step_6_usage_reporting/report.py:91]: The purpose of this change is to promote labeling adoption. For Marketplace-defined labels, SaaS app (data plane) have the data and populates them without requiring user inputs. I think we may make this clear to partners somehow, for example, adding comments or calling a function etc. Does it make sense?

File: java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java (hunk 1 above)
Comment (arshan-alam) [reply java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java:53]: Yup

File: java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java (hunk 2 above)
Comment (arshan-alam) [reply java/impl/src/main/java/com/example/saas/step6usagereporting/Report.java:67]: Yup formatting.

**Review by tangbiao — APPROVED** [review]

**Pull request merged**